roles/pool/src/lib/mod.rs

```diff
@@ -32,17 +32,32 @@ impl PoolSv2 {
         let coinbase_output_result = get_coinbase_output(&config);
         let coinbase_output_len = coinbase_output_result?.len() as u32;
         let tp_authority_public_key = config.tp_authority_public_key;
-        TemplateRx::connect(
-            config.tp_address.parse().unwrap(),
-            s_new_t,
-            s_prev_hash,
-            r_solution,
-            r_message_recv_signal,
-            status::Sender::Upstream(status_tx.clone()),
-            coinbase_output_len,
-            tp_authority_public_key,
-        )
-        .await?;
+
+        loop {
```

Comment:
Here you loop through the template provider connection regardless of the error, not only if TP is not up. Instead the loop should be added here https://github.com/stratum-mining/stratum/blob/main/roles/pool/src/lib/template_receiver/mod.rs#L48